x/pool-incentives/keeper/grpc_query_test.go

```diff
@@ -178,39 +173,15 @@ func (s *KeeperTestSuite) TestIncentivizedPools() {
 		{
 			desc:                 "Normal case",
 			poolCreated:          true,
-			weights:              []osmomath.Int{osmomath.NewInt(100), osmomath.NewInt(200), osmomath.NewInt(300)},
-			expectedRecordLength: 3,
-		},
-		{
-			desc:                 "Perpetual",
-			poolCreated:          true,
-			weights:              []osmomath.Int{osmomath.NewInt(100), osmomath.NewInt(200), osmomath.NewInt(300)},
-			perpetual:            true,
-			expectedRecordLength: 3,
-		},
-		{
-			desc:                 "Non Perpetual",
-			poolCreated:          true,
-			weights:              []osmomath.Int{osmomath.NewInt(100), osmomath.NewInt(200), osmomath.NewInt(300)},
-			nonPerpetual:         true,
-			expectedRecordLength: 0,
-		},
-		{
-			desc:                 "Concentrated case, no pool migration link",
-			poolCreated:          true,
-			clPoolWithGauge:      true,
-			clGaugeWeight:        osmomath.NewInt(400),
-			weights:              []osmomath.Int{osmomath.NewInt(100), osmomath.NewInt(200), osmomath.NewInt(300)},
-			expectedRecordLength: 3, // Note we expect 3 instead of 4 here, since the pool migration link is not setup.
+			weights:              []osmomath.Int{osmomath.NewInt(100), osmomath.NewInt(200), osmomath.NewInt(300), osmomath.NewInt(400)},
+			expectedRecordLength: 4, // three for gamm pool, one for cl pool
 		},
 		{
-			desc:                   "Concentrated case, setup pool migration link",
-			poolCreated:            true,
-			clPoolWithGauge:        true,
-			clGaugeWeight:          osmomath.NewInt(400),
-			weights:                []osmomath.Int{osmomath.NewInt(100), osmomath.NewInt(200), osmomath.NewInt(300)},
-			setupPoolMigrationLink: true,
-			expectedRecordLength:   4,
+			desc:                     "Perpetual Group Gauge",
```

Comment:
Could we also the non-perpetual group being skipped?

Reply:
We physically cant add non-perpetual anythings to groupRecords